<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Yusha Liu" />


<title>Exploration of inconsistency between scatter plot and GLM fit</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">poisson_mash_sc</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/YushaLiu/poisson_mash_sc">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Exploration of inconsistency between scatter plot and GLM fit</h1>
<h4 class="author">Yusha Liu</h4>
<h4 class="date">2021-4-28</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-04-29
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>poisson_mash_sc/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210330code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210330)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210330code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210330)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomYushaLiupoissonmashsctree6356a56c51d6332bd8ab924bd41c490d6266ace6targetblank6356a56a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/YushaLiu/poisson_mash_sc/tree/6356a56c51d6332bd8ab924bd41c490d6266ace6" target="_blank">6356a56</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomYushaLiupoissonmashsctree6356a56c51d6332bd8ab924bd41c490d6266ace6targetblank6356a56a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/YushaLiu/poisson_mash_sc/tree/6356a56c51d6332bd8ab924bd41c490d6266ace6" target="_blank">6356a56</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    analysis/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  code/plot_by_gene.R
    Untracked:  code/plot_by_gene_v2.R

Unstaged changes:
    Deleted:    analysis/application_Ly6C+.Rmd
    Deleted:    analysis/application_Ly6C-.Rmd
    Deleted:    output/Report_B_cells.html
    Deleted:    output/Report_Dendritic_cells.html
    Deleted:    output/Report_Ly6C_neg_monocytes.html
    Deleted:    output/Report_Ly6C_pos_monocytes.html

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Alox5ap.Rmd</code>) and HTML (<code>docs/Alox5ap.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/YushaLiu/poisson_mash_sc/blob/6356a56c51d6332bd8ab924bd41c490d6266ace6/analysis/Alox5ap.Rmd" target="_blank">6356a56</a>
</td>
<td>
yushaliu
</td>
<td>
2021-04-29
</td>
<td>
minor edits
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/YushaLiu/poisson_mash_sc/00729a1513d5af6dcdd3ad0e4de5940c3c4fbbeb/docs/Alox5ap.html" target="_blank">00729a1</a>
</td>
<td>
yushaliu
</td>
<td>
2021-04-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/YushaLiu/poisson_mash_sc/blob/e2ec4aaa234825156bfc9528dc7614ef84975d1c/analysis/Alox5ap.Rmd" target="_blank">e2ec4aa</a>
</td>
<td>
yushaliu
</td>
<td>
2021-04-28
</td>
<td>
update reference line in scatter plot
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/YushaLiu/poisson_mash_sc/e7ad94020e26aec66abca6d87a5c1e817c98f772/docs/Alox5ap.html" target="_blank">e7ad940</a>
</td>
<td>
yushaliu
</td>
<td>
2021-04-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/YushaLiu/poisson_mash_sc/blob/862e95d8762d4504e96ee4537816b0fa9db8734f/analysis/Alox5ap.Rmd" target="_blank">862e95d</a>
</td>
<td>
yushaliu
</td>
<td>
2021-04-28
</td>
<td>
inconsistency
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="model-setup" class="section level2">
<h2>Model setup</h2>
<p>Suppose there are <span class="math inline">\(j=1, \dots, J\)</span> genes and <span class="math inline">\(i=1, \dots, N\)</span> cells. The observed single cell count matrix <span class="math inline">\(Y\)</span> is <span class="math inline">\(N \times J\)</span>, with its <span class="math inline">\((i, j)\)</span> element <span class="math inline">\(Y_{ij}\)</span> denoting the count of gene <span class="math inline">\(j\)</span> in cell <span class="math inline">\(i\)</span>.</p>
<p>We assume that the <span class="math inline">\(N\)</span> cells come from <span class="math inline">\(r=1, \dots, R\)</span> conditions, with <span class="math inline">\(n_r\)</span> cells (indexed by <span class="math inline">\(\mathcal{S}_r \subset \{1, \dots, N\}\)</span>) coming from condition <span class="math inline">\(r\)</span>. To compare expression levels across multiple conditions, we collapse the single cell count matrix <span class="math inline">\(Y\)</span> into a condition level count matrix <span class="math inline">\(X\)</span>, which is a <span class="math inline">\(J \times R\)</span> matrix with its <span class="math inline">\((j, r)\)</span> element <span class="math inline">\(X_{jr} = \sum_{i \in \mathcal{S}_r} Y_{ij}\)</span>.</p>
<p>Let <span class="math inline">\(s_i\)</span> denote the size factor of cell <span class="math inline">\(i\)</span>, which can be calculated by taking the sum (or equivalently, mean) of counts over all genes in cell <span class="math inline">\(i\)</span>, or using other <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0947-7">more robust methods</a>. Let <span class="math inline">\(s_r = \sum_{i \in \mathcal{S}_r} s_i\)</span> denote the size factor of condition <span class="math inline">\(r\)</span>.</p>
<p>Ignoring the random effect term and unwanted variation for now, we assume the following model on collapsed data <span class="math inline">\(X\)</span>: <span class="math display">\[\begin{align}
(1). \quad\quad &amp; X_{jr} \; \sim Pois (s_r \lambda_{jr}), \\
(2). \quad\quad &amp; \log \left(\lambda_{jr} \right) \; = \tilde{\beta}_{jr} = \mu_j + \beta_{jr}. \\
\end{align}\]</span></p>
<p>Here <span class="math inline">\(\lambda_{jr}\)</span> is a gene-specific, condition-specific intensity parameter, <span class="math inline">\(s_r\)</span> is condition-specific and accounts for different sequencing depths across conditions. We decompose <span class="math inline">\(\log(\lambda_{jr})\)</span> for gene <span class="math inline">\(j\)</span> and condition <span class="math inline">\(r\)</span> into 2 parts: (i) <span class="math inline">\(\mu_j\)</span> which models the gene-specific underlying mean shared by all conditions, (ii) <span class="math inline">\(\beta_{jr}\)</span> which models the gene-specific, condition-specific effect.</p>
</div>
<div id="estimate-of-log-fold-change-from-scatter-plot" class="section level2">
<h2>Estimate of log-fold-change from scatter plot</h2>
<p>Given Eq(1), the MLE estimate of <span class="math inline">\(\lambda_{jr}\)</span> is <span class="math inline">\(\hat{\lambda}_{jr} = \frac{X_{jr}}{s_r}\)</span>. Assuming <span class="math inline">\(\lambda_{jr} = \lambda_j\)</span> for <span class="math inline">\(\forall \; r\)</span>, the MLE estimate of <span class="math inline">\(\lambda_j\)</span> is <span class="math inline">\(\hat{\lambda}_j = \frac{\sum_r X_{jr}}{\sum_r s_r}\)</span>. Therefore, the log-fold-change seen from the scatter plot is: <span class="math display">\[\begin{align}
(3). \quad\quad \log(\frac{\hat{\lambda}_{jr}}{\hat{\lambda}_j}) = \log(\frac{X_{jr}/s_r}{\sum_r X_{jr}/ \sum_r s_r}) = \log(\frac{X_{jr}}{(\sum_r X_{jr}) \frac{s_r}{\sum_r{s_r}}}) = \log(\frac{\text{observed}}{\text{expected}}).
\end{align}\]</span></p>
</div>
<div id="estimate-of-log-fold-change-from-log-linear-model" class="section level2">
<h2>Estimate of log-fold-change from log-linear model</h2>
<p>Given the log-linear model Eq(2), the log-fold-change between condition <span class="math inline">\(r\)</span> and the average is: <span class="math display">\[\begin{align}
(4). \quad\quad \hat{\beta}_{jr} - \frac{1}{R}\sum_r\hat{\beta}_{jr} = \hat{\tilde{\beta}}_{jr} - \frac{1}{R}\sum_r \hat{\tilde{\beta}}_{jr} = \log(\hat{\lambda}_{jr}) - \frac{1}{R}\sum_r \log(\hat{\lambda}_{jr}) \neq \log(\hat{\lambda}_{jr}) - \log(\hat{\lambda}_j),
\end{align}\]</span> since <span class="math inline">\(\frac{1}{R}\sum_r \log(\hat{\lambda}_{jr}) \neq \log(\hat{\lambda}_j)\)</span> unless <span class="math inline">\(\frac{X_{jr}}{s_r}\)</span> is the same for all <span class="math inline">\(r\)</span>. This explains why the log-fold-change suggested by the scatter plot is different from the estimate of <span class="math inline">\(\beta_{jr} - \frac{1}{R}\sum_r\beta_{jr}\)</span>.</p>
</div>
<div id="example-alox5ap" class="section level2">
<h2>Example: Alox5ap</h2>
<pre class="r"><code>cols.trt &lt;- readRDS(&quot;../applications_sc/B_cells/condition_colors.Rds&quot;)
trts &lt;- names(cols.trt)

### swap the color code of IL1b and IL3 for clearer display of IL1b
cols.trt2 &lt;- cols.trt
cols.trt2[which(trts==&quot;IL1b&quot;)] &lt;- cols.trt[which(trts==&quot;IL3&quot;)]
cols.trt2[which(trts==&quot;IL3&quot;)] &lt;- cols.trt[which(trts==&quot;IL1b&quot;)]
cols.trt &lt;- cols.trt2

### load in condition-level count data
data &lt;- readRDS(&quot;../applications_sc/Neutrophils/data_jr.Rds&quot;)
data.jr &lt;- data$data.jr
idx.gene &lt;- which(rownames(data.jr)==&quot;Alox5ap&quot;)
x &lt;- data.jr[idx.gene,]
s &lt;- data$s</code></pre>
</div>
<div id="alox5ap-log-fold-change-from-the-scatter-plot" class="section level2">
<h2>Alox5ap: log-fold-change from the scatter plot</h2>
<pre class="r"><code>loglambda &lt;- log(sum(x)/sum(s))
loglambda.jr &lt;- log(x/s)

lfc1 &lt;- loglambda.jr - loglambda
names(lfc1) &lt;- paste0(trts, &quot;-mean&quot;)

# log fold change
round(lfc1, 3)</code></pre>
<pre><code> Ctrl_2-mean   CCL20-mean   CXCL1-mean   CCL22-mean   CXCL5-mean   CCL11-mean 
      -0.428       -0.356       -0.329       -0.411       -0.395       -0.320 
   CCL4-mean   CCL17-mean    CCL5-mean  CXCL13-mean  CXCL10-mean   CXCL9-mean 
      -0.499       -0.307       -0.408       -0.412       -0.439       -0.406 
 CXCL12-mean    GCSF-mean    MCSF-mean   GMCSF-mean    IFNg-mean    IL10-mean 
      -0.335        0.083       -0.396       -0.362       -0.599       -0.331 
IL12p70-mean   IL17a-mean    IL13-mean    IL15-mean   IL17f-mean    IL22-mean 
      -0.523       -0.341       -0.315       -0.305       -0.349       -0.303 
   IL18-mean    IL1a-mean     IL2-mean     IL3-mean    IL1b-mean    IL23-mean 
      -0.419        0.603       -0.517       -0.400        0.162       -0.347 
   IL21-mean    IL33-mean    IL25-mean    IL34-mean    IL27-mean   IL36a-mean 
      -0.355       -0.403       -0.469       -0.323       -0.372       -0.409 
    IL4-mean     IL6-mean     IL5-mean     IL7-mean     IL9-mean    IL11-mean 
      -0.401       -0.333       -0.380       -0.375       -0.427       -0.398 
   TGFb-mean    CCL2-mean    CCL3-mean    TSLP-mean 
      -0.378       -0.362       -0.302       -0.440 </code></pre>
<pre class="r"><code># scatter plots comparing observed and expected counts for each treatment
data.scatter &lt;- data.frame(trt=trts, observed=x, expected=sum(x)*s/sum(s))
par(mfrow=c(1,1))

xmin &lt;- pmin(min(c(data.scatter$observed, data.scatter$expected)) - 5, 0)
xmax &lt;- max(c(data.scatter$observed, data.scatter$expected)) + 10
plot(data.scatter$observed, data.scatter$expected, col=cols.trt, pch=19, xlim=c(xmin, xmax), ylim=c(xmin, xmax),  
     xlab=&quot;observed count&quot;, ylab=&quot;expected count&quot;, main=&quot;Alox5ap&quot;)
text(x=data.scatter$observed + (xmax-xmin)/50, y=data.scatter$expected + (xmax-xmin)/50, labels=trts, col=cols.trt, cex=0.6)
abline(a=0, b=1, col=&quot;gray&quot;)</code></pre>
<p><img src="figure/Alox5ap.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/YushaLiu/poisson_mash_sc/blob/e7ad94020e26aec66abca6d87a5c1e817c98f772/docs/figure/Alox5ap.Rmd/unnamed-chunk-2-1.png" target="_blank">e7ad940</a>
</td>
<td>
yushaliu
</td>
<td>
2021-04-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="alox5ap-log-fold-change-from-the-log-linear-model" class="section level2">
<h2>Alox5ap: log-fold-change from the log-linear model</h2>
<pre class="r"><code>lfc2 &lt;- loglambda.jr - mean(loglambda.jr)
names(lfc2) &lt;- paste0(trts, &quot;-mean&quot;)

# log fold change
round(lfc2, 3)</code></pre>
<pre><code> Ctrl_2-mean   CCL20-mean   CXCL1-mean   CCL22-mean   CXCL5-mean   CCL11-mean 
      -0.084       -0.011        0.016       -0.067       -0.051        0.024 
   CCL4-mean   CCL17-mean    CCL5-mean  CXCL13-mean  CXCL10-mean   CXCL9-mean 
      -0.155        0.037       -0.064       -0.068       -0.094       -0.062 
 CXCL12-mean    GCSF-mean    MCSF-mean   GMCSF-mean    IFNg-mean    IL10-mean 
       0.009        0.427       -0.052       -0.018       -0.254        0.013 
IL12p70-mean   IL17a-mean    IL13-mean    IL15-mean   IL17f-mean    IL22-mean 
      -0.179        0.003        0.029        0.039       -0.005        0.041 
   IL18-mean    IL1a-mean     IL2-mean     IL3-mean    IL1b-mean    IL23-mean 
      -0.075        0.947       -0.173       -0.055        0.506       -0.003 
   IL21-mean    IL33-mean    IL25-mean    IL34-mean    IL27-mean   IL36a-mean 
      -0.011       -0.059       -0.125        0.021       -0.028       -0.065 
    IL4-mean     IL6-mean     IL5-mean     IL7-mean     IL9-mean    IL11-mean 
      -0.057        0.011       -0.036       -0.031       -0.083       -0.054 
   TGFb-mean    CCL2-mean    CCL3-mean    TSLP-mean 
      -0.034       -0.018        0.042       -0.096 </code></pre>
<pre class="r"><code># load in glm fit with poisson link
res.glm &lt;- readRDS(&quot;../applications_sc/Neutrophils/glm_poisson.Rds&quot;)
round(res.glm$est.pre.shrink[idx.gene,], 3)</code></pre>
<pre><code> Ctrl_2-mean   CCL20-mean   CXCL1-mean   CCL22-mean   CXCL5-mean   CCL11-mean 
      -0.084       -0.011        0.016       -0.067       -0.051        0.024 
   CCL4-mean   CCL17-mean    CCL5-mean  CXCL13-mean  CXCL10-mean   CXCL9-mean 
      -0.155        0.037       -0.064       -0.068       -0.094       -0.062 
 CXCL12-mean    GCSF-mean    MCSF-mean   GMCSF-mean    IFNg-mean    IL10-mean 
       0.009        0.427       -0.052       -0.018       -0.254        0.013 
IL12p70-mean   IL17a-mean    IL13-mean    IL15-mean   IL17f-mean    IL22-mean 
      -0.179        0.003        0.029        0.039       -0.005        0.041 
   IL18-mean    IL1a-mean     IL2-mean     IL3-mean    IL1b-mean    IL23-mean 
      -0.075        0.947       -0.173       -0.055        0.506       -0.003 
   IL21-mean    IL33-mean    IL25-mean    IL34-mean    IL27-mean   IL36a-mean 
      -0.011       -0.059       -0.125        0.021       -0.028       -0.065 
    IL4-mean     IL6-mean     IL5-mean     IL7-mean     IL9-mean    IL11-mean 
      -0.057        0.011       -0.036       -0.031       -0.083       -0.054 
   TGFb-mean    CCL2-mean    CCL3-mean    TSLP-mean 
      -0.034       -0.018        0.042       -0.096 </code></pre>
<pre class="r"><code># load in poisson mash without ruv
res.gene &lt;- readRDS(&quot;../applications_sc/Neutrophils/poisson_mash_no_ruv/Alox5ap.Rds&quot;)
round(res.gene$PosteriorMean, 3)</code></pre>
<pre><code> Ctrl_2-mean   CCL20-mean   CXCL1-mean   CCL22-mean   CXCL5-mean   CCL11-mean 
      -0.117        0.024        0.028       -0.025       -0.007        0.017 
   CCL4-mean   CCL17-mean    CCL5-mean  CXCL13-mean  CXCL10-mean   CXCL9-mean 
      -0.123        0.087       -0.076       -0.014       -0.098       -0.022 
 CXCL12-mean    GCSF-mean    MCSF-mean   GMCSF-mean    IFNg-mean    IL10-mean 
       0.005        0.459       -0.060       -0.003       -0.202       -0.034 
IL12p70-mean   IL17a-mean    IL13-mean    IL15-mean   IL17f-mean    IL22-mean 
      -0.218        0.010        0.051        0.029       -0.018        0.033 
   IL18-mean    IL1a-mean     IL2-mean     IL3-mean    IL1b-mean    IL23-mean 
      -0.084        0.897       -0.158       -0.073        0.535       -0.013 
   IL21-mean    IL33-mean    IL25-mean    IL34-mean    IL27-mean   IL36a-mean 
      -0.004       -0.107       -0.182       -0.017       -0.125       -0.095 
    IL4-mean     IL6-mean     IL5-mean     IL7-mean     IL9-mean    IL11-mean 
      -0.012       -0.013       -0.009       -0.063        0.002       -0.070 
   TGFb-mean    CCL2-mean    CCL3-mean    TSLP-mean 
      -0.043       -0.074        0.056       -0.075 </code></pre>
<p>As can be seen above, the estimates of log-fold-change are highly consistent between</p>
<p>1. <span class="math inline">\(\hat{\beta}_{jr} - \frac{1}{R}\sum_r\hat{\beta}_{jr}\)</span> given by Eq(4);</p>
<p>2. estimate of <span class="math inline">\(\beta_{jr} -\frac{1}{R}\sum_r\beta_{jr}\)</span> based on GLM fit to the cell-level count data with Poisson link;</p>
<p>3. posterior mean of <span class="math inline">\(\beta_{jr} -\frac{1}{R}\sum_r\beta_{jr}\)</span> given by Poisson mash without ruv.</p>
</div>
<div id="alox5ap-scatter-plot-with-updated-reference-line" class="section level2">
<h2>Alox5ap: scatter plot with updated reference line</h2>
<p>Let <span class="math inline">\(c_j = \log(\hat{\lambda}_{j}) - \frac{1}{R}\sum_r\log(\hat{\lambda}_{jr})\)</span>, and <span class="math inline">\(\tilde{X}_{jr} = (\sum_r X_{jr}) \frac{s_r}{\sum_r{s_r}}\)</span>, i.e., the expected count of <span class="math inline">\(X_{jr}\)</span> in the scatter plot. Following Eqs(3)-(4), we have <span class="math display">\[\begin{align}
(5). \quad\quad \log(\frac{\tilde{X}_{jr}}{X_{jr}}) = \log(\hat{\lambda}_j) - \log(\hat{\lambda}_{jr}) = c_j - \log(\hat{\lambda}_{jr}) + \frac{1}{R}\sum_r\log(\hat{\lambda}_{jr}) = c_j - \hat{\beta}_{jr} + \frac{1}{R}\sum_r\hat{\beta}_{jr}. \\
\end{align}\]</span></p>
<p>Assuming <span class="math inline">\(\beta_{jr} - \frac{1}{R}\sum_r\beta_{jr}=0\)</span> for <span class="math inline">\(r\)</span>, the updated reference line in the scatter plot should be <span class="math display">\[\begin{align}
(6). \quad\quad Y = X\exp(c_j).\\
\end{align}\]</span></p>
<pre class="r"><code># c.j for Alox5ap
c.j &lt;- loglambda - mean(loglambda.jr)
c.j</code></pre>
<pre><code>[1] 0.3441261</code></pre>
<pre class="r"><code># scatter plots comparing observed and expected counts for each treatment
xmin &lt;- pmin(min(c(data.scatter$observed, data.scatter$expected)) - 5, 0)
xmax &lt;- max(c(data.scatter$observed, data.scatter$expected)) + 10
plot(data.scatter$observed, data.scatter$expected, col=cols.trt, pch=19, xlim=c(xmin, xmax), ylim=exp(c.j)*c(xmin, xmax),  
     xlab=&quot;observed count&quot;, ylab=&quot;expected count&quot;, main=&quot;Alox5ap&quot;)
text(x=data.scatter$observed + (xmax-xmin)/50, y=data.scatter$expected + (xmax-xmin)/50, labels=trts, col=cols.trt, cex=0.6)
abline(a=0, b=exp(c.j), col=&quot;gray&quot;)</code></pre>
<p><img src="figure/Alox5ap.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/YushaLiu/poisson_mash_sc/blob/00729a1513d5af6dcdd3ad0e4de5940c3c4fbbeb/docs/figure/Alox5ap.Rmd/unnamed-chunk-4-1.png" target="_blank">00729a1</a>
</td>
<td>
yushaliu
</td>
<td>
2021-04-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.2 (2019-12-12)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.6      rprojroot_2.0.2 digest_0.6.27   later_1.0.0    
 [5] R6_2.5.0        git2r_0.27.1    magrittr_2.0.1  evaluate_0.14  
 [9] stringi_1.4.6   rlang_0.4.10    fs_1.4.1        promises_1.1.0 
[13] whisker_0.4     rmarkdown_2.3   tools_3.6.2     stringr_1.4.0  
[17] glue_1.4.2      httpuv_1.5.2    xfun_0.17       yaml_2.2.1     
[21] compiler_3.6.2  htmltools_0.4.0 knitr_1.30     </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
