<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Yusha Liu" />


<title>Description of poisson mash ruv model</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">poisson_mash_sc</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/YushaLiu/poisson_mash_sc">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Description of poisson mash ruv model</h1>
<h4 class="author">Yusha Liu</h4>
<h4 class="date">2021-4-22</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-04-22
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>poisson_mash_sc/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomYushaLiupoissonmashsctree531070c4b7f320846487516d55bca0c56a654fc7targetblank531070ca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/YushaLiu/poisson_mash_sc/tree/531070c4b7f320846487516d55bca0c56a654fc7" target="_blank">531070c</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomYushaLiupoissonmashsctree531070c4b7f320846487516d55bca0c56a654fc7targetblank531070ca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/YushaLiu/poisson_mash_sc/tree/531070c4b7f320846487516d55bca0c56a654fc7" target="_blank">531070c</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  code/plot_by_gene.R

Unstaged changes:
    Deleted:    analysis/application_Ly6C+.Rmd
    Deleted:    analysis/application_Ly6C-.Rmd
    Deleted:    output/Report_B_cells.html
    Deleted:    output/Report_Dendritic_cells.html
    Deleted:    output/Report_Ly6C_neg_monocytes.html
    Deleted:    output/Report_Ly6C_pos_monocytes.html

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/model.Rmd</code>) and HTML (<code>docs/model.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/YushaLiu/poisson_mash_sc/blob/531070c4b7f320846487516d55bca0c56a654fc7/analysis/model.Rmd" target="_blank">531070c</a>
</td>
<td>
yushaliu
</td>
<td>
2021-04-22
</td>
<td>
upload presentation results
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="model-setup" class="section level2">
<h2>Model setup</h2>
<p>Suppose there are <span class="math inline">\(j=1, \dots, J\)</span> genes and <span class="math inline">\(i=1, \dots, N\)</span> cells. The observed single cell count matrix <span class="math inline">\(Y\)</span> is <span class="math inline">\(N \times J\)</span>, with its <span class="math inline">\((i, j)\)</span> element <span class="math inline">\(Y_{ij}\)</span> denoting the count of gene <span class="math inline">\(j\)</span> in cell <span class="math inline">\(i\)</span>.</p>
<p>We assume that the <span class="math inline">\(N\)</span> cells come from <span class="math inline">\(r=1, \dots, R\)</span> conditions, with <span class="math inline">\(n_r\)</span> cells (indexed by <span class="math inline">\(\mathcal{S}_r \subset \{1, \dots, N\}\)</span>) coming from condition <span class="math inline">\(r\)</span>. To compare expression levels across multiple conditions, we collapse the single cell count matrix <span class="math inline">\(Y\)</span> into a condition level count matrix <span class="math inline">\(X\)</span>, which is a <span class="math inline">\(J \times R\)</span> matrix with its <span class="math inline">\((j, r)\)</span> element <span class="math inline">\(X_{jr} = \sum_{i \in \mathcal{S}_r} Y_{ij}\)</span>.</p>
<p>Let <span class="math inline">\(s_i\)</span> denote the size factor of cell <span class="math inline">\(i\)</span>, which can be calculated by taking the sum (or equivalently, mean) of counts over all genes in cell <span class="math inline">\(i\)</span>, or using other <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0947-7">more robust methods</a>. Let <span class="math inline">\(s_r = \sum_{i \in \mathcal{S}_r} s_i\)</span> denote the size factor of condition <span class="math inline">\(r\)</span>.</p>
</div>
<div id="poisson-mash-ruv-removing-unwanted-variation" class="section level2">
<h2>Poisson mash ruv (removing unwanted variation)</h2>
<p>In poisson mash ruv, we assume the following hierarchical model on aggregated condition-level count matrix <span class="math inline">\(X\)</span>: <span class="math display">\[\begin{align}
(1). \quad\quad\quad &amp; X_{jr} \sim \; Pois (s_r \lambda_{jr}), \\
(2). \quad\quad\quad &amp; \log \left(\lambda_{jr} \right) = \; \mu_j + \beta_{jr} + \eta_{jr} + \sum_{d=1}^D \rho_{rd} f_{jd}, \\
(3). \quad\quad\quad &amp; \beta_j \sim \; \sum_{k,l} \pi_{kl} N(0, w_l U_k)  \quad \text{where} \;\; \sum_{k,l} \pi_{kl} = 1,  \\
(4). \quad\quad\quad &amp; \eta_j \sim \; N(0, \psi_j^2 I_R). \\
\end{align}\]</span></p>
<p>In Eq(1), <span class="math inline">\(\lambda_{jr}\)</span> is a gene-specific, condition-specific intensity parameter, <span class="math inline">\(s_r\)</span> is condition-specific and accounts for different sequencing depths across conditions.</p>
<p>In Eq(2), we decompose <span class="math inline">\(\log(\lambda_{jr})\)</span> for gene <span class="math inline">\(j\)</span> and condition <span class="math inline">\(r\)</span> into 4 parts: (i) <span class="math inline">\(\mu_j\)</span> which models the gene-specific underlying mean shared by all conditions, (ii) <span class="math inline">\(\beta_{jr}\)</span> which models the gene-specific, condition-specific effect, (iii) <span class="math inline">\(\eta_{jr}\)</span> which models the individual sample effect, and (iv) <span class="math inline">\(\sum_{d} \rho_{rd} f_{jd}\)</span> which represents the bias caused by unwanted variation.</p>
<p>In Eq(3), we assign <span class="math inline">\(\beta_{jr}\)</span> the mash prior (a mixture of MVNs) that can achieve adaptive shrinkage and allow arbitrary patterns of correlations among conditions. <span class="math inline">\(U_k (k=1, \dots, K)\)</span> include canonical prior covariances <span class="math inline">\(e_r e_r&#39;\)</span> that model condition-specific effects, and data-driven prior covariances that model effect-sharing patterns among conditions, and <span class="math inline">\(w_l (l=1, \dots, L)\)</span> are scaling parameters that model varying sizes of effects. In fitting poisson mash ruv, we are mainly interested in estimation and inference on <span class="math inline">\(\beta_{jr}\)</span>.</p>
<p>In Eq(4), we assign the individual sample effect term <span class="math inline">\(\eta_{jr}\)</span> a MVN prior with zero mean and gene-specific covariance <span class="math inline">\(\psi_j^2 I_R\)</span>, which can account for over-dispersion of count data. To see this, we can integrate out the term <span class="math inline">\(\eta_{jr}\)</span> and get <span class="math display">\[\begin{align}
\mathbb{E} \left[X_{jr} \right] =  s_r \exp \left(\mu_j + \beta_{jr} + \sum_{d=1}^D \rho_{rd} f_{jd} + \frac{1}{2} \psi_j^2 \right), 
\end{align}\]</span> <span class="math display">\[\begin{align}
\text{Var} \left(X_{jr} \right) = s_r \exp \left(\mu_j + \beta_{jr} + \sum_{d=1}^D \rho_{rd} f_{jd} + \frac{1}{2} \psi_j^2 \right) + s_r^2 \left(\exp(\psi_j^2) - 1 \right) \exp \left(2 \mu_j + 2 \beta_{jr} + 2 \sum_{d=1}^D \rho_{rd} f_{jd} + \psi_j^2 \right), 
\end{align}\]</span></p>
<p>which shows that <span class="math inline">\(\text{Var} \left(X_{jr} \right)\)</span> is larger than <span class="math inline">\(\mathbb{E} \left[X_{jr} \right]\)</span> when <span class="math inline">\(\psi_j &gt; 0\)</span>.</p>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
