<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Yusha Liu" />


<title>Application of poisson mash and competing methods to simulated single cell data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">poisson_mash_sc</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/YushaLiu/poisson_mash_sc">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Application of poisson mash and competing methods to simulated single cell data</h1>
<h4 class="author">Yusha Liu</h4>
<h4 class="date">2021-3-30</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-03-31
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>poisson_mash_sc/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210330code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210330)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210330code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210330)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomYushaLiupoissonmashsctree29e00976f397df86396aa0f7c15f993eceef182dtargetblank29e0097a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/YushaLiu/poisson_mash_sc/tree/29e00976f397df86396aa0f7c15f993eceef182d" target="_blank">29e0097</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomYushaLiupoissonmashsctree29e00976f397df86396aa0f7c15f993eceef182dtargetblank29e0097a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/YushaLiu/poisson_mash_sc/tree/29e00976f397df86396aa0f7c15f993eceef182d" target="_blank">29e0097</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/simulations.Rmd</code>) and HTML (<code>docs/simulations.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/YushaLiu/poisson_mash_sc/blob/29e00976f397df86396aa0f7c15f993eceef182d/analysis/simulations.Rmd" target="_blank">29e0097</a>
</td>
<td>
yushaliu
</td>
<td>
2021-03-31
</td>
<td>
Update results for poisson mash ruv on scdata
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="description-of-approaches" class="section level2">
<h2>Description of approaches</h2>
<div id="mash-based-approaches" class="section level3">
<h3>MASH-based approaches:</h3>
<p>1. poisson mash: fit poisson mash with gene-specific means and random effects to the count data while not accounting for unwanted variation.</p>
<p>2. poisson mash ruv: fit poisson mash with gene-specific means and random effects to the count data while accounting for unwanted variation. For ruv, the <span class="math inline">\(J \times D\)</span> factor matrix is estimated using <a href="https://cran.r-project.org/web/packages/glmpca/index.html">GLMPCA</a> while protecting the condition-specific effects.</p>
<p>3. mash: fit mash with gene-specific means and random effects to the gene expression data normalized using <a href="https://satijalab.org/seurat/">Seurat</a> while not accounting for unwanted variation. We set <span class="math inline">\(\alpha = 1\)</span>, i.e., the EZ model.</p>
<p>4. mash ruv: fit mash with gene-specific means and random effects to the normalized data while accounting for unwanted variation. For ruv, the <span class="math inline">\(J \times D\)</span> factor matrix is estimated using <a href="https://github.com/willwerscheid/flashier">flashier</a> while protecting the condition-specific effects. We set <span class="math inline">\(\alpha = 1\)</span>, i.e., the EZ model.</p>
<p>5. mash cb: fit mash with no common baseline to the normalized data using <a href="https://stephenslab.github.io/mashr/articles/intro_mashnobaseline.html">mashr</a>, where the reference is the mean over all conditions. We set <span class="math inline">\(\alpha = 1\)</span>, i.e., the EZ model. The implementation is similar to method 3 but without the random effect term. Also, the selection of “strong” subset of genes in the ED step is conducted differently.</p>
</div>
<div id="benchmark-approaches" class="section level3">
<h3>Benchmark approaches:</h3>
<p>We also implemented the best-performing methods identified by this comprehensive <a href="https://www.nature.com/articles/nmeth.4612.pdf?origin=ppub">benchmark paper</a>, which evaluated 36 approaches that have been developed for DE analysis on bulk RNA-seq data or scRNA-seq data. We only considered methods that allow for comparison across multiple (<span class="math inline">\(&gt;2\)</span>) conditions, and performed an ANOVA-like test for differences between any of the conditions. With each approach, we obtained a gene-wise p-value adjusted for multiple comparisons based on FDR.</p>
<p>6. <a href="https://link.springer.com/protocol/10.1007/978-1-4939-3578-9_19">edgeR-QLF-DetRate</a>: fit a quasi-likelihood negative binomial generalized log-linear model to the count data which includes the cellular detection rate (the fraction of genes that are detected with nonzero counts) as a covariate in the model.</p>
<p>7. <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0844-5">MAST-cpm-DetRate</a>: fit a two-part generalized linear model to test DE, with one component modeling the discrete expression rate of each gene across cells and the other component modeling the conditional continuous expression level. The cellular detection rate is included as a covariate in the model.</p>
<p>8. <a href="http://www.bioconductor.org/packages/devel/bioc//manuals/limma/man/limma.pdf">limma-trend</a>: fit limma-trend to the <span class="math inline">\(\log_2\)</span> transformed CPM data.</p>
<p>9. <a href="https://en.wikipedia.org/wiki/Kruskal–Wallis_one-way_analysis_of_variance">Kruskal-Wallis test</a>: perform Kruskal-Wallis test for DE analysis, which is a non-parametric analogue of ANOVA.</p>
</div>
</div>
<div id="simulation-scenarios" class="section level2">
<h2>Simulation scenarios</h2>
<p>To simulate single cell data, we started with a subset of the single cell cytokines data. We first randomly permuted the cells with respect to the condition labels to create null data, then used thinning to add condition effects <span class="math inline">\(\beta_{jr} \sim \sum_k \pi_k MVN(0, U_k)\)</span> to a subset of genes, where the prior covariances <span class="math inline">\(U_k\)</span>’s are learned from the poisson mash fit to a subset of B cells in the original dataset, with the effect size (in terms of <span class="math inline">\(\max_r \mid \beta_{jr}\mid\)</span>) further scaled to be realistic. We started with the following subsets of the real data, and removed genes with fewer than 25 counts summed over all cells.</p>
<p>Setting 1: a subset of B cells with 8,358 genes and 2,096 cells from 25 cytokine treatments.</p>
<p>Setting 2: a larger subset of B cells with 10,691 genes and 15,705 cells from 25 cytokine treatments.</p>
<p>Setting 3: a subset of cells from “Ctrl2” with 9,041 genes and 2,182 cells from 8 cell types. Since the DE analysis results across cell types, which are themselves identified based on gene expression profiles, are dubious without appropriate correction, we only assessed the performance of methods on simulated null data under this scenario.</p>
<p>We simulate 20 single cell datasets for each scenario.</p>
</div>
<div id="simulation-results-for-null-data" class="section level2">
<h2>Simulation results for null data</h2>
<div id="setting-1" class="section level3">
<h3>Setting 1</h3>
<pre class="r"><code>### read in result summary
num_effects &lt;- readRDS(&quot;../simulations_sc/setting1/sim_null_summary.Rds&quot;)

### number of false positives
num_effects</code></pre>
<pre><code>                 rep1 rep2 rep3 rep4 rep5 rep6 rep7 rep8 rep9 rep10 rep11 rep12
edgeRQLFDetRate    83   88   75   90   78   93   65   89   92    75    93   102
MASTcpmDetRate      1    1    0    2    0    0    0    0    1     0     2     1
limma-trend         1    1    0    3    0    0    0    0    1     0     2     0
KW-test             0    0    0    2    0    0    0    0    0     0     2     0
mash               11   11   13   11    3    7    7    5   12     5     7    11
mash cb            16   11   21   14    8   10   16   13   14     9    12    15
mash ruv           12   12   13   11    7   11    9   11   16     8    11    13
poisson mash       13   26   11   34   31   42   16   23   22    20    24    54
poisson mash ruv   11   25    7   36    9    9   10    8   10    13    26     9
                 rep13 rep14 rep15 rep16 rep17 rep18 rep19 rep20
edgeRQLFDetRate     93    67    80    90    84    90    91    92
MASTcpmDetRate       1     0     0     0     1     1     0     1
limma-trend          0     0     0     0     0     1     0     0
KW-test              0     0     0     1     1     1     0     1
mash                 3    12     6    11     9    16    14    15
mash cb              9    13    12    11    15    19    16    14
mash ruv             6    19     9    13    12    21    17    17
poisson mash        16    12    17    24    38    11    25    16
poisson mash ruv     8     9     9     9    20     8    23     6</code></pre>
</div>
<div id="setting-2" class="section level3">
<h3>Setting 2</h3>
<pre class="r"><code>### read in result summary
num_effects &lt;- readRDS(&quot;../simulations_sc/setting2/sim_null_summary.Rds&quot;)

### number of false positives
num_effects</code></pre>
<pre><code>                 rep1 rep2 rep3 rep4 rep5 rep6 rep7 rep8 rep9 rep10 rep11 rep12
edgeRQLFDetRate    35   25   23   24   26   30   24   21   23    22    32    27
MASTcpmDetRate      3    3    0    1    3    0    1    1    4     1     1     1
limma-trend         6    4    0    0    3    0    3    1    7     0     1     0
KW-test             2    1    0    0    3    0    2    1    6     0     1     1
mash               19   25   23   15   20    7   22   30   23    35    20    19
mash cb            24   24   23   31   25   12   32   28   21    35    29    19
mash ruv           20   22   24   13   21    9   23   29   20    36    23    22
poisson mash       13   10   19   81   24   23    9   24   23    18    22    40
poisson mash ruv   13    8    7    7    7   11    7   16    8     5     5     6
                 rep13 rep14 rep15 rep16 rep17 rep18 rep19 rep20
edgeRQLFDetRate     42    26    24    25    19    19    24    23
MASTcpmDetRate       0     0     3     2     1     1     1     1
limma-trend          0     0     3     2     1     1     0     1
KW-test              0     0     3     2     1     0     3     1
mash                15    17    21    16    14    33    21    36
mash cb             18    22    15    17    23    31    20    40
mash ruv            13    14    21    15    17    33    22    39
poisson mash        26    28    33    51    54    20    22    23
poisson mash ruv    11    11     8    10    10    10     9    10</code></pre>
</div>
<div id="setting-3" class="section level3">
<h3>Setting 3</h3>
<pre class="r"><code>### read in result summary
num_effects &lt;- readRDS(&quot;../simulations_sc/setting3/sim_null_summary.Rds&quot;)

### number of false positives
num_effects</code></pre>
<pre><code>                 rep1 rep2 rep3 rep4 rep5 rep6 rep7 rep8 rep9 rep10 rep11 rep12
mash                0    0    0    0    0    0    0    0    0     0     0     0
mash cb            30   23   52   11   27   48   39   46   35    16    29    20
mash ruv            0    0    0    0    0    0    0    0    0     0     0     0
poisson mash       84   10   46   73  110  188   26  140   15    73    47    36
poisson mash ruv    9    3   11   13    5   14    6    5    5     6     8     2
                 rep13 rep14 rep15 rep16 rep17 rep18 rep19 rep20
mash                 0     0     0     0     0     0     0     0
mash cb             12    22    25    16    90    60    22    48
mash ruv             0     0     0     0     0     0     0     0
poisson mash        96   114    74   146   143   100    33   120
poisson mash ruv     6     1     6     1     2     2     4    13</code></pre>
</div>
</div>
<div id="effect-sharing-patterns-across-conditions-for-non-null-data" class="section level2">
<h2>Effect-sharing patterns across conditions for non-null data</h2>
<pre class="r"><code>cols.trt &lt;- readRDS(&quot;../simulations_sc/setting2/condition_colors.Rds&quot;)
ulist.sim &lt;- readRDS(&quot;../simulations_sc/setting2/ulist_sim.rds&quot;)

for (k in 1:length(ulist.sim$ulist)){
  v &lt;- pmax(ulist.sim$ulist[[k]], 0)
  
  barplot(v/v[which.max(abs(v))], names = names(cols.trt), cex.names = 0.8, las = 2, main = paste0(&quot;Effect-sharing pattern: u&quot;, k), col = cols.trt)
}</code></pre>
<p><img src="figure/simulations.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/simulations.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/simulations.Rmd/unnamed-chunk-4-3.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.2 (2019-12-12)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.6      rprojroot_2.0.2 digest_0.6.27   later_1.0.0    
 [5] R6_2.5.0        git2r_0.27.1    magrittr_2.0.1  evaluate_0.14  
 [9] stringi_1.4.6   rlang_0.4.10    fs_1.4.1        promises_1.1.0 
[13] whisker_0.4     rmarkdown_2.3   tools_3.6.2     stringr_1.4.0  
[17] glue_1.4.2      httpuv_1.5.2    xfun_0.17       yaml_2.2.1     
[21] compiler_3.6.2  htmltools_0.4.0 knitr_1.30     </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
