---
title: "Application of poisson mash ruv to Ly6C- monocytes in single cell cytokines data"
author: "Yusha Liu"
header-includes:
date: "2021-4-14"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Summary
We apply poisson mash ruv and mash to a subset of Ly6C- monocytes from a single cell cytokines dataset, where the conditions represent different types of cytokine treatments ($R=46$). This dataset includes expression data of $J=10,286$ genes from $3,855$ cells. Genes with fewer than $25$ counts summed over all cells are removed. 
```{r warning=FALSE, message=FALSE}
library(pheatmap)
library(gridExtra)

cols.trt <- readRDS("../applications_sc/B_cells/condition_colors.Rds")
trts <- names(cols.trt)

```


## Running ED step
To estimate the data-driven prior covariances, we fit poisson mash ruv without scaling parameters to a subset of genes that are identified to be differentially expressed across conditions by a conditional multinomial goodness-of-fit test. Compared to the previous implementation, the $R$ canonical prior covariances modeling the condition-specific effects are also included in the ED step with their scaling parameters to be estimated. We expect that in the presence of these canonical prior covariances, we can better estimate the data-driven prior covariances which capture the effect-sharing patterns across conditions in the data.
```{r warning=FALSE, message=FALSE}
## load in the ED results
fit.ed <- readRDS("../applications_sc/Ly6C-/pois_mash_ruv_ed_v3.Rds")

## running time per iteration in ED
fit.ed$runtime/length(fit.ed$ELBO)
                      
## look at the ELBO
plot(fit.ed$ELBO, xlab = "niter", ylab = "elbo", main="fitting ED step")

```


## Adding epsilon2 to the diagonals of rank-1 prior covariances
For genes whose posterior weights concentrate on rank-1 prior covariances (mostly data-driven ones), the poisson mash ruv fit often produces [unrealistically tight credible intervals](https://yushaliu.github.io/poisson_mash_sc/application_B_cells.html#A_few_examples) for the condition-specific effects. To alleviate this, we added a small positive number, denoted by $\epsilon^2$, to the diagonals of the rank-1 prior covariances that receive non-negligible prior weights (> 1\%) in the ED step. To determine the size of $\epsilon^2$ that should be added, we refitted the ED step by replacing these rank-1 prior covariances $u_k u_k'$ with $u_k u_k' + \epsilon^2 I$ (here $u_k$ is normalized such that its largest element is 1), and plotted the ELBO as a function of $\epsilon^2$.   
```{r warning=FALSE, message=FALSE}
### load in the ED fit corresponding to different values of epsilon2
log2_epsilon2_list <- seq(log2(1e-6), log2(1), length.out = 20)
epsilon2_list <- 2^log2_epsilon2_list
ELBOs <- rep(NA, 20)

for(l in 1:20){
  fit.ed.l <- readRDS(paste0("../applications_sc/Ly6C-/epsilon2_estimation/pois_mash_ruv_ed_", l, ".Rds"))
  ELBOs[l] <- fit.ed.l$ELBO[length(fit.ed.l$ELBO)]
}


### look at the epsilon2 and ELBOs
formatC(epsilon2_list, format = "e", digits = 1)
ELBOs


### plot ELBO as a function of epsilon2
plot(1:20, ELBOs, type="p", xaxt="n", xlab="epsilon2", ylab="ELBO", main="ELBO as a function of epsilon2 in the ED step")
axis(1, at=1:20, labels=formatC(epsilon2_list, format = "e", digits = 0), las=2)

```
The ELBO peaks at $\epsilon^2 = 7.8e-5$, and the largest ELBO is -236561.6. For $\epsilon^2 = 1.4e-3$, ELBO is -236705.5. For $\epsilon^2 = 1.3e-2$, ELBO is -236823.4. However, we found that adding an $\epsilon^2 \approx 1e-4$ doesn't seem to effectively solve the tight error bar issue. 


## Running poisson mash ruv
We then fit poisson mash ruv to the entire dataset, with $\epsilon^2 = 1e-2$.
```{r warning=FALSE, message=FALSE}
## load in the poisson mash ruv fit
res <- readRDS("../applications_sc/Ly6C-/pois_mash_ruv_fit_v5.Rds")

## running time
res$runtime

## look at ELBO
plot(res$ELBO, xlab = "niter", ylab = "elbo", main="Fitting poisson mash ruv")

## calculate prior weights
pi.mat <- matrix(res$pi, ncol=length(res$wlist), byrow=TRUE)

rownames(pi.mat) <- c(names(res$Ulist), names(res$ulist))
colnames(pi.mat) <- paste0("w=", round(res$wlist, 3))
pheatmap(pi.mat, cluster_rows=FALSE, cluster_cols=FALSE, fontsize_row = 6, fontsize_col=6, 
         main="Estimated weights of prior covariances in poisson mash ruv")

rowSums(pi.mat, 3)

```



## Estimate of data-driven covariances from ED step
```{r warning=FALSE, message=FALSE}
### Estimate of top eigenvectors of the data-driven covariance initialized by rank-5 PCA
eig.tPCA <- eigen(fit.ed$Ulist[[1]])
pve.tPCA <- eig.tPCA$values/sum(eig.tPCA$values)

for (k in 1:2){
  v <- eig.tPCA$vectors[,k]
  
  barplot(v/v[which.max(abs(v))], names = names(cols.trt), cex.names = 0.5, las = 2, 
          main = paste0(names(fit.ed$Ulist)[1], ": weight ", round(rowSums(pi.mat)[1],3), 
                        ":\nEigenvector ", k, " (pve = ", round(pve.tPCA[k],3), ")"), col = cols.trt)
}


### Estimate of top eigenvectors of rank-1 data-driven covariance matrices
for (k in 1:length(fit.ed$ulist)){
  if(rowSums(pi.mat)[k+2] > 5e-2 & sum(fit.ed$ulist[[k]]!=0)>0){
    v <- as.numeric(fit.ed$ulist[[k]])
    barplot(v/v[which.max(abs(v))], names = names(cols.trt), cex.names = 0.5, las = 2, col = cols.trt,
            main = paste0(names(fit.ed$ulist)[k], ": weight ", round(rowSums(pi.mat)[k+2],3)))    
  }
}

```



## Comparison to poisson mash ruv fit without adding epsilon2
We also implemented poisson mash ruv without adding $\epsilon^2$ to the diagonals of rank-1 prior covariances, and compared the identified DE genes to the results of the poisson mash ruv fit with $\epsilon^2=1e-2$. We also implemented mash without common baseline to the gene expression data normalized using [Seurat](https://satijalab.org/seurat/). For all implementations, the reference is the mean over all 46 conditions. 
```{r warning=FALSE, message=FALSE}
### load in the different versions of model fit
pois.ruv.v1 <- readRDS("../applications_sc/Ly6C-/pois_mash_ruv_posterior_v3.Rds")
pois.ruv.v2 <- readRDS("../applications_sc/Ly6C-/pois_mash_ruv_posterior_v5.Rds")
mash.cb <- readRDS("../applications_sc/Ly6C-/mash_common_baseline.Rds")

### poisson mash ruv without epsilon2 added
idx1.pois <- which(apply(pois.ruv.v1$lfsr, 1, min) < 0.05)
length(idx1.pois)

### poisson mash ruv with epsilon2=1e-2 added
idx2.pois <- which(apply(pois.ruv.v2$lfsr, 1, min) < 0.05)
length(idx2.pois)

### look at the overlap of the two sets of DE genes 
sum(idx2.pois %in% idx1.pois)

### mash without common baseline
idx.mash.cb <- mashr::get_significant_results(mash.cb)
length(idx.mash.cb)

```


## A few examples
```{r warning=FALSE, message=FALSE}
### load in the plotting function
source("../applications_sc/B_cells/plot_by_gene_v3.R")

### load in the condition-level count data
data <- readRDS("../applications_sc/Ly6C-/data_jr.Rds")
data.jr <- data$data.jr
s <- data$s

plot_by_gene(data=data.jr, s=s, res.pois1=pois.ruv.v1, res.pois2=pois.ruv.v2, res.mash=mash.cb$result, genename="Stat1", cols.trt=cols.trt)
plot_by_gene(data=data.jr, s=s, res.pois1=pois.ruv.v1, res.pois2=pois.ruv.v2, res.mash=mash.cb$result, genename="Gbp7", cols.trt=cols.trt)
plot_by_gene(data=data.jr, s=s, res.pois1=pois.ruv.v1, res.pois2=pois.ruv.v2, res.mash=mash.cb$result, genename="Gbp3", cols.trt=cols.trt)
plot_by_gene(data=data.jr, s=s, res.pois1=pois.ruv.v1, res.pois2=pois.ruv.v2, res.mash=mash.cb$result, genename="Klra2", cols.trt=cols.trt)
plot_by_gene(data=data.jr, s=s, res.pois1=pois.ruv.v1, res.pois2=pois.ruv.v2, res.mash=mash.cb$result, genename="Fgl2", cols.trt=cols.trt)
plot_by_gene(data=data.jr, s=s, res.pois1=pois.ruv.v1, res.pois2=pois.ruv.v2, res.mash=mash.cb$result, genename="Irf1", cols.trt=cols.trt)
plot_by_gene(data=data.jr, s=s, res.pois1=pois.ruv.v1, res.pois2=pois.ruv.v2, res.mash=mash.cb$result, genename="Serpina3g", cols.trt=cols.trt)
plot_by_gene(data=data.jr, s=s, res.pois1=pois.ruv.v1, res.pois2=pois.ruv.v2, res.mash=mash.cb$result, genename="Psmb9", cols.trt=cols.trt)
plot_by_gene(data=data.jr, s=s, res.pois1=pois.ruv.v1, res.pois2=pois.ruv.v2, res.mash=mash.cb$result, genename="Psmb8", cols.trt=cols.trt)

```
